lib/_stream_writable.js

```diff
@@ -367,6 +362,12 @@ function writeOrBuffer(stream, state, chunk, encoding, cb) {
     doWrite(stream, state, false, len, chunk, encoding, cb);
   }
 
+  const ret = state.length < state.highWaterMark;
+
+  // We must ensure that previous needDrain will not be reset to false.
+  if (!ret)
+    state.needDrain = true;
```

Comment:
I would add a few comments about this change here.

Reply:
I'm not sure what kind of comment?